caffeine@patapon.info/preferences/timerPage.js

```diff
@@ -0,0 +1,339 @@
+        this.enableCustomTimerRow.connect('notify::active', (widget) => {
+            this._settings.set_boolean(this._settingsKey.USE_CUSTOM_DURATION, widget.get_active());
+            this._activeCustomvalue();
+        });
+        this.sliderTimer.connect('change-value',
+            (widget) => this._updateTimerDuration(widget.get_value()));
+        this.resetCustomTimerButton.connect('clicked', () => {
+            this._updateCustomDurationFromIndex(this.sliderTimer.get_value());
+        });
+        this.shortTimerSelector.connect('notify::value', (widget) => {
+            this._updateDurationVarian(widget.get_value(), 0);
+            this._updateResetButtonState();
+            // Control hierarchy of custom duration
+            const shortValue = this.shortTimerSelector.get_value();
+            const mediumValue = this.mediumTimerSelector.get_value();
+            if (shortValue >= mediumValue) {
+                this.mediumTimerSelector.set_value(shortValue + 60);
+            }
+        });
+        this.mediumTimerSelector.connect('notify::value', (widget) => {
+            this._updateDurationVarian(widget.get_value(), 1);
+            this._updateResetButtonState();
+            // Control hierarchy of custom duration
+            const shortValue = this.shortTimerSelector.get_value();
+            const mediumValue = this.mediumTimerSelector.get_value();
+            const longValue = this.longTimerSelector.get_value();
+            if (mediumValue <= shortValue) {
+                this.shortTimerSelector.set_value(mediumValue - 60);
+            }
+            if (mediumValue >= longValue) {
+                this.longTimerSelector.set_value(mediumValue + 60);
+            }
+        });
+        this.longTimerSelector.connect('notify::value', (widget) => {
+            this._updateDurationVarian(widget.get_value(), 2);
+            this._updateResetButtonState();
+            // Control hierarchy of custom duration
+            const mediumValue = this.mediumTimerSelector.get_value();
+            const longValue = this.longTimerSelector.get_value();
+            if (longValue <= mediumValue) {
+                this.mediumTimerSelector.set_value(longValue - 60);
+            }
+        });
+    }
+
+    _isCustomValueSet() {
+        const shortValue = this.shortTimerSelector.get_value();
+        const mediumValue = this.mediumTimerSelector.get_value();
+        const longValue = this.longTimerSelector.get_value();
+        const durationIndex = this.sliderTimer.get_value();
+        let isCustomed = false;
+        if (shortValue !== parseInt(TIMERS_DURATION[durationIndex][0]) * 60) {
+            isCustomed = true;
+        }
+        if (mediumValue !== parseInt(TIMERS_DURATION[durationIndex][1]) * 60) {
+            isCustomed = true;
+        }
+        if (longValue !== parseInt(TIMERS_DURATION[durationIndex][2]) * 60) {
+            isCustomed = true;
+        }
+        return isCustomed;
```

Comment:
```suggestion
        const selectors = [this.shortTimerSelector, this.mediumTimerSelector, this.longTimerSelector];
        const durationIndex = this.sliderTimer.get_value();
        let isCustom = false;

        selectors.forEach((selector, i) => {
            if (selector.get_value() !== parseInt(TIMERS_DURATION[durationIndex][i]) * 60) {
                isCustom = true;
            }
        });

        return isCustom;
```

Untested, but would something like this be better? I wasn't going to nit-pick this because it's not huge, but if you've already got to fix the other issue this might be worth looking at too.

Reply:
I will.